src/App.jsx

```diff
@@ -1,11 +1,24 @@
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import AddTodoForm from './AddTodoForm'
 import './App.css'
 import TodoList from './TodoList'
 
+function useSemiPersistentState(){
+
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')));
```

Comment:
What would the `todoList` be set to if the localStorage is empty? Hint: it will be related to an error that you would see when your components try to `map` out the `todoList` state variable.